### ISSUE TITLE
Tests Execution hangs on Azure pipeline after upgrading Angular to version 15

### ISSUE THREAD

**Issue by shavezhameed** [cypress-io/cypress#25432]
### Current behavior

Current angular Version: 15
Current Cypress Version: 12.3

Since the upgrade of angular to version 15 my tests in the pipeline are getting stuck until the timeout occurs from the pipeline.
Locally the tests are running fine.

Is this an angular issue or Cypress Issue?

### Test code to reproduce

Here are the logs from the pipeline : 

```
2023-01-11T01:10:26.1913158Z     ✔  All specs passed!                        00:03        1        1        -        -        -  
2023-01-11T01:10:26.1916367Z 
2023-01-11T02:00:58.8321079Z ##[error]The operation was canceled.
2023-01-11T02:00:58.8371715Z ##[section]Finishing: Execute cypress tests
2023-01-11T02:00:59.8858781Z ##[section]Starting: Publish Test Results
2023-01-11T02:00:59.8882457Z ==============================================================================
2023-01-11T02:00:59.8882764Z Task         : Publish Test Results
2023-01-11T02:00:59.8882936Z Description  : Publish test results to Azure Pipelines
2023-01-11T02:00:59.8883137Z Version      : 2.210.0
2023-01-11T02:00:59.8883286Z Author       : Microsoft Corporation
2023-01-11T02:00:59.8883476Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/test/publish-test-results
2023-01-11T02:00:59.8883751Z ==============================================================================
2023-01-11T02:01:10.9959362Z [command]/usr/bin/dotnet --version
2023-01-11T02:01:10.9959755Z 7.0.100
2023-01-11T02:01:13.0343893Z Result Attachments will be stored in LogStore
2023-01-11T02:01:13.2050448Z Run Attachments will be stored in LogStore
2023-01-11T02:01:13.3208050Z ##[section]Async Command Start: Publish test results
2023-01-11T02:01:13.6457602Z Publishing test results to test run '49616'.
2023-01-11T02:01:13.6491897Z TestResults To Publish 1, Test run id:49616
2023-01-11T02:01:13.6545302Z Test results publishing 1, remaining: 0. Test run id: 49616
2023-01-11T02:01:15.1895355Z Published Test Run: [url]
2023-01-11T02:01:15.3275396Z ##[section]Async Command End: Publish test results
2023-01-11T02:01:15.3277173Z ##[section]Finishing: Publish Test Results
2023-01-11T02:01:15.7865011Z ##[section]Starting: Checkout xxxxx@master to s
2023-01-11T02:01:15.7871965Z ==============================================================================
2023-01-11T02:01:15.7872302Z Task         : Get sources
2023-01-11T02:01:15.7872453Z Description  : Get sources from a repository. Supports Git, TfsVC, and SVN repositories.
2023-01-11T02:01:15.7872731Z Version      : 1.0.0
2023-01-11T02:01:15.7873157Z Author       : Microsoft

2023-01-11T02:01:15.7874196Z ==============================================================================
2023-01-11T02:01:16.3413288Z Cleaning any cached credential from repository:
2023-01-11T02:01:16.3422485Z ##[section]Finishing: Checkout 
2023-01-11T02:01:16.9549052Z ##[section]Starting: Finalize Job
2023-01-11T02:01:16.9595259Z Cleaning up task key
2023-01-11T02:01:16.9597713Z Start cleaning up orphan processes.
2023-01-11T02:01:17.0061533Z Terminate orphan process: pid (3907) (ng run app-e2e:e2e)
2023-01-11T02:01:17.0098636Z Terminate orphan process: pid (3970) (esbuild)
2023-01-11T02:01:17.0187171Z ##[section]Finishing: Finalize Job
2023-01-11T02:01:17.0242594Z ##[section]Finishing: Job
```

### Cypress Version

12.3

### Node version

18.13

### Operating System

Windows

**Comment by sergiubologa** [comment]
I have a similar issue, although I'm not using Angular. I have an open discussion here: https://github.com/cypress-io/cypress/discussions/25431 but no answer yet

**Comment by sergiubologa** [comment]
possible duplicate of https://github.com/cypress-io/cypress/issues/24896

**Comment by sergiubologa** [comment]
Can you try using the cypress version `10.10.0` and see if it still reproduces for you?

**Comment by astone123** [comment]
@shavezhameed could you also run Cypress with [debug logs](https://docs.cypress.io/guides/references/troubleshooting#Print-DEBUG-logs) in your pipeline and then post those logs here so that we can see what gets printed before it hangs? Thanks

**Comment by shavezhameed** [comment]
@astone123 please find below the logs 

2023-01-18T12:11:44.297Z cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 60354 }
2023-01-18T12:11:44.297Z cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 60308 }
2023-01-18T12:11:44.298Z cypress:server:socket-base socket-disconnecting transport close
2023-01-18T12:11:44.305Z cypress:server:socket-base socket-disconnect transport close
2023-01-18T12:11:44.306Z cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 41792 }

================================================================================

  (Run Finished)
tput: No value for $TERM and no -T specified


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  login.cy.ts                              00:02        1        1        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        00:02        1        1        -        -        -  

2023-01-18T12:11:44.373Z cypress:server:cypress about to exit with code 0
2023-01-18T12:11:44.373Z cypress:server:browsers browsers.kill called with no active instance
2023-01-18T12:11:44.373Z cypress:proxy:http:util:prerequests metrics: { browserPreRequestsReceived: 43, proxyRequestsReceived: 12, immediatelyMatchedRequests: 7, unmatchedRequests: 2, unmatchedPreRequests: 27 }
2023-01-18T12:11:44.483Z cypress:cli child event fired { event: 'exit', code: 0, signal: null }
2023-01-18T12:11:44.483Z cypress:cli Stopping Xvfb
2023-01-18T12:11:44.484Z cypress:cli child event fired { event: 'close', code: 0, signal: null }


This is where it hangs.

**Comment by shavezhameed** [comment]
> Can you try using the cypress version `10.10.0` and see if it still reproduces for you?

I did, same issue.

**Comment by ZachJW34** [comment]
I noticed in your logs that there is an orphaned process `esbuild`, are you using the experimental esbuild angular builder?

I cannot replicate your Azure pipeline, but if possible can you try running two tests to determine where the issue lies?

- Simple Angular 14 app with a Cypress e2e test on Azure pipeline
- Simple Angular 15 app with a Cypress e2e test on Azure pipeline

If you are using the `esbuild` Angular builder, can you switch to the default build process (webpack) and see if anything changes? From your description it looks like the issue lies with the Angular upgrade

**Comment by shavezhameed** [comment]
@ZachJW34 we are using webpack only, The issue doesnt seem to be there in angular 13 but need to try angular 14.

**Comment by gmfun** [comment]
I see this issue after migrating from angular 14 to angular 15 and cypress 11 to cypress 12. but cypress 10.10.1 works 


update: Sorry, for wrong reporting. Cypress 10.10.x does not work. @sergiubologa 

I am also using nx and running using nx command

**Comment by ZachJW34** [comment]
How are the tests being executed? Are you invoking Cypress directly (npx cypress run) or using something like `@cypress/schematic` or Nx DevTools to launch Cypress (ng e2e my-app)?

**Comment by shavezhameed** [comment]
We use Nx Devtools, So I upgraded to their latest versions too, but still
no luck.

**Comment by webberig** [comment]
I was also running into the same issue with one of our projects. The problem was caused by the presence of https://github.com/manfredsteyer/ngx-build-plus. We use this builder to alter the webpack configuration.

This was working fine up to Angular 13, it broke while upgrading to Angular 14.

When I run the cypress tests with the default @angular-devkit builder, everything is back to normal. Hope this helps...

**Comment by ZachJW34** [comment]
@shavezhameed could you provide a minimal reproduction of your Nx + Angular + Cypress application? I don't have access to Azure pipelines but I could try it against a GitHub action or some other free CI service.

You might also want to try using another browser. Also, the debug logs you posted only show a brief glimpse into what might be going on, could you share the entirety of the logs following the documentation [here](https://docs.cypress.io/guides/references/troubleshooting#Print-DEBUG-logs)?

**Comment by shavezhameed** [comment]
@ZachJW34 I have tried all different browsers.

Once the test execution is finished this is what the logs say :

    ✖  2 of 24 failed (8%)                      18:26      121      110       11        -        -  

tput: No value for $TERM and no -T specified
--------------------------------------------------------------------------------

  Having trouble debugging your CI failures?
  
  Record your runs to Cypress Cloud to watch video recordings for each test, 
  debug failing and flaky tests, and integrate with your favorite tools.

  >> https://on.cypress.io/cloud-get-started

tput: No value for $TERM and no -T specified
--------------------------------------------------------------------------------
2023-02-03T09:36:53.979Z cypress:server:cypress about to exit with code 11
2023-02-03T09:36:53.980Z cypress:server:browsers browsers.kill called with no active instance
2023-02-03T09:36:54.009Z cypress:proxy:http:util:prerequests metrics: { browserPreRequestsReceived: 4306, proxyRequestsReceived: 1880, immediatelyMatchedRequests: 739, unmatchedRequests: 14, unmatchedPreRequests: 2440 }
2023-02-03T09:36:54.010Z cypress:server:util:socket_allowed allowed socket closed, removing { localPort: 57842 }
2023-02-03T09:36:54.070Z cypress:cli child event fired { event: 'exit', code: 11, signal: null }
2023-02-03T09:36:54.070Z cypress:cli Stopping Xvfb
2023-02-03T09:36:54.073Z cypress:cli child event fired { event: 'close', code: 11, signal: null }
##[error]The operation was canceled.
Finishing: Execute cypress tests


Also, i will try to provide the minimal version of the application meanwhile.

**Comment by shavezhameed** [comment]
@webberig Thanks, I will try to look in this direction as well and update you.

**Comment by marktnoonan** [comment]
Hi @shavezhameed, since it's been more than a week since the last response I'm going to close this for now. Please comment again if you have an update and we can re-open at that time.
